### PR TITLE
upgraded react-native to 0.59.0-rc.2

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

This was necessary for Android 64-bit ARM support, related to this issue: 

https://github.com/facebook/react-native/issues/2814

This could probably use some iOS testing for verification.

### Checklist:

- [ ] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any displayed ETH addresses with [`formattedAddress`](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/user.js#L15-L17)
- [ ] Wrap any new text/strings for translation
- [ ] Run `npm run translations` if there are any changes to translated strings
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### CODE REVIEW

File: origin-mobile/src/components/transaction-item.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ class TransactionItem extends Component {
       <TouchableHighlight onPress={handlePress}>
         <View style={[ styles.listItem, styles.completed, style ]}>
           {!picture && <View style={{ ...styles.thumbnail, ...styles.imageless }} />}
-          {picture && <Image source={{ uri: picture.url || picture }} style={styles.thumbnail} />}
+          {picture !== '' && <Image source={{ uri: picture.url || picture }} style={styles.thumbnail} />}
```
Comment (micahalcorn) [inline origin-mobile/src/components/transaction-item.js]: @mikeshultz did you find a problem resulting from this loose checking for a `picture` value?

Comment (micahalcorn) [reply origin-mobile/src/components/transaction-item.js]: For example, if `picture` is `undefined` or `null` for some reason, won't we end up with both the `View` and the `Image`?

Comment (mikeshultz) [reply origin-mobile/src/components/transaction-item.js]: Yeah, this was hasty, but it was to tackle an actual bug.  When the user had no stored avatar and wasn't setting one, `picture` was an empty string.  And for whatever reason RN was trying to render that empty string causing errors.  I don't really have a full picture of what its values could be, however.

How about an explicit boolean conversion, instead?  e.g. `Boolean(picture) && [...]`

**Comment by mikeshultz** [comment]
Added explicit bool conversions.  Also, one of the checks was just wrong.  Thanks for pointing out this hasty change.

**Comment by micahalcorn** [comment]
Great, thanks @mikeshultz 🙏 I'm actually about to add placeholder images in those cases, so that should take care of it too.

**Comment by micahalcorn** [comment]
c560b74 was used for the latest TestFlight build (0.2.9). @crazybuster said I could merge this. 😬

**Pull request merged**